eslint.config.mjs

```diff
@@ -259,6 +260,35 @@ const config = createConfig([
       sourceType: 'module',
     },
   },
+  // Tailwind ESLint
+  {
+    files: [
+      'packages/design-tokens/stories/**',
+      'packages/design-system-react-native/src/**',
+      'apps/storybook-react-native/stories/**',
+      'packages/design-system-react/src/**',
+      'apps/storybook-react/stories/**',
+    ],
+    plugins: {
+      tailwindcss: tailwind,
+    },
+    rules: {
+      'tailwindcss/classnames-order': 'error',
+      'tailwindcss/enforces-negative-arbitrary-values': 'error',
+      'tailwindcss/enforces-shorthand': 'error',
+      'tailwindcss/no-arbitrary-value': 'off', // There are legitimate reasons to use arbitrary values but we should specifically error on static colors
+      'tailwindcss/no-custom-classname': 'error',
+      'tailwindcss/no-contradicting-classname': 'error',
+      'tailwindcss/no-unnecessary-arbitrary-value': 'error',
+    },
+    settings: {
+      tailwindcss: {
+        callees: ['tw', 'twMerge', 'twClassName'],
+        config: 'apps/storybook-react/tailwind.config.js',
+        classRegex: '^class(Name)?$',
+      },
+    },
+  },
```

Comment:
We should have good coverage for both React Native and React libraries, but a few points to note:

* We're still using the React Tailwind config for linting, so some classes not supported by `twrnc` may not be flagged. This will be addressed in [[this ticket](https://github.com/MetaMask/metamask-design-system/issues/737)](https://github.com/MetaMask/metamask-design-system/issues/737).
* Classnames in our `*constant.ts` files aren't being caught by the plugin. I've tried several regex approaches without success and have opened an issue [[here](https://github.com/MetaMask/metamask-design-system/issues/763)](https://github.com/MetaMask/metamask-design-system/issues/763).